src/mappings/permissionedExchange.ts

```diff
@@ -0,0 +1,139 @@
+import { AcalaEvmEvent } from "@subql/acala-evm-processor";
+import { ExchangeOrderSentEvent, OrderSettledEvent, TradeEvent } from "@subql/contract-sdk/typechain/PermissionedExchange";
+import assert from "assert";
+import { OrderStatus } from "../types";
+
+import { PermissionedExchange__factory } from '@subql/contract-sdk';
+import FrontierEthProvider from './ethProvider';
+import { EXCHANGE_DIST_ADDRESS, getUpsertAt } from "./utils";
+import { Order, Trade, Trader } from "../types";
+import { BigNumber } from 'ethers';
+
+const { ACTIVE, INACTIVE } = OrderStatus;
+
+function calculateTradeAmount(
+    totalTradeAmount: bigint,
+    event: AcalaEvmEvent<TradeEvent['args']>
+): bigint {
+    const { tokenGive, tokenGet, amountGive, amountGet } = event.args;
+    const getIsKSQT = tokenGet === 'kSQT';
+    const giveIsKSQT = tokenGive === 'kSQT';
+
+    if(giveIsKSQT && getIsKSQT) return totalTradeAmount;
+
+    const tradeAmountBN = BigNumber.from(totalTradeAmount);
+
+    if(giveIsKSQT) return tradeAmountBN.add(amountGive).toBigInt();
+    if(getIsKSQT) return tradeAmountBN.sub(amountGet).toBigInt();
+
+    return totalTradeAmount;
+}
+
+async function createTrade(
+    orderId: BigNumber,
+    sender: string,
+    event: AcalaEvmEvent<TradeEvent['args']>
+): Promise<void> {
+    const { tokenGive, tokenGet, amountGive, amountGet } = event.args;
+
+    const trade = Trade.create({
```

Comment:
As all the values are from order perspective, so need to make it different for trade
```
    const trade = Trade.create({
        id: `${orderId.toString()}:${event.transactionHash}`,
        tokenGive: tokenGet,
        tokenGet: tokenGive,
        amountGive:  amountGive.toBigInt(),
        amountGet: amountGet.toBigInt(),
        senderId: sender
    });
    ```